### ISSUE TITLE
onTapAnnotation click event fired together with onTapListener Mapbox

### ISSUE THREAD

**Issue by maxchris100** [mapbox/mapbox-maps-flutter#199]
onTapAnnotation click event fired together with onTapListener Mapbox

**Comment by nathan-poncet** [comment]
Yes, and when you have several annotations in the same place, the marker you click on is selected at random.

**Comment by tiyberius** [comment]
I am experiencing this as well @nathan-poncet. I'm currently working around it with a Debounce timer.

`Timer? _debounce;

class AnnotationClickListener extends OnPointAnnotationClickListener {
  @override
  void onPointAnnotationClick(PointAnnotation annotation) {
    if (_debounce?.isActive ?? false) _debounce?.cancel();
    _debounce = Timer(const Duration(milliseconds: 20), () {
      sw.Location? location = annotationIdToLocation[annotation.id];
      if (location != null) {
        router.pushNamed(
          ...
        );
      }
    });
  }
}
`
